Write StatefulSet List, Patch & DeleteCollection Test +3 endpoints

riaankleinhans opened the pull request:
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it:**
This PR adds a test to test the following untested endpoints:
- patchAppsV1NamespacedStatefulSet
- listAppsV1StatefulSetForAllNamespaces
- deleteAppsV1CollectionNamespacedStatefulSet

**Which issue(s) this PR fixes:**
Fixes #100727
**Testgrid Link:**
[Testgrid](https://k8s-testgrid.appspot.com/sig-release-master-blocking#gce-cos-master-default&width=5&include-filter-by-regex=should%20list,%20patch%20and%20delete%20a%20collection%20of%20StatefulSets&graph-metrics=test-duration-minutes)


**Special notes for your reviewer:**
Adds +3 endpoint test coverage (good for conformance)

**Does this PR introduce a user-facing change?:**
```
NONE

```

**Release note:**
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:**
```
NONE

```
/sig apps
/sig testing
/sig architecture
/area conformance

Riaankl commented:
/triage accepted

Riaankl commented:
/assign @ravisantoshgudimetla @soltysh @smarterclayton 
This test follows much the same pattern used in the other Apps e2e & Conformance test to clean up the last technical debt.

ravisantoshgudimetla reviewed (approved):
Thank you for working on this @Riaankl 

/lgtm

soltysh reviewed (approved):
/priority backlog
/approve

Riaankl commented:
/hold
Need to rebase
/assign @heyste

heyste committed:
Create List, Patch & DeleteCollection StatefulSet Test

Co-Authored-By: Riaan Kleinhans <riaan@ii.coop>

e2e test validates the following 3 extra endpoints

- patchAppsV1NamespacedStatefulSet
- listAppsV1StatefulSetForAllNamespaces
- deleteAppsV1CollectionNamespacedStatefulSet

Riaankl commented:
/unhold

hh commented:
/lgtm
